core/comment_events.js

```diff
@@ -243,6 +329,13 @@ Blockly.Events.CommentDelete = function(comment) {
     return;  // Blank event to be populated by fromJson.
   }
   Blockly.Events.CommentDelete.superClass_.constructor.call(this, comment);
+  this.xy = comment.getXY();
```

Comment:
Update type annotation on the `comment` param to include ScratchBlocksComments.

When I figure out how to make typedefs work, we can typedef those two together for use in all of these places.